rfcs/0146-meta-categories.md

```diff
@@ -0,0 +1,298 @@
+   easier to create such categories using Nix language when compared to other
+   software collections.
+
+3. Superfluous
+
+   It can be argued that there are other ways to discover similar or related
+   package sets, like Repology.
+
+   However, this argument is a bit circular, because e.g. the classification
+   shown by Repology effectively replicates the classification done by the many
+   software collections in its catalog. Therefore, relying in Repology merely
+   transfers the question to external sources.
+
+   Further it becomes more pronounced when we take into account the fact Nixpkgs
+   is top 1 of most Repology statistics. The expected outcome, therefore, should
+   be precisely the opposite: Nixpkgs being _the_ source of structured metainfo
+   for other software collections.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Do nothing
+
+   This will exacerbate the problems already listed.
+
+2. Ignore/nuke the categorization completely
+
+   This is an alternative worthy of some consideration. After all,
+   categorization is not without its problems, as shown above. Removing or
+   ignoring classification removes all problems.
+
+   However, there are good reasons to keep the categorization:
+
+   - The complete removal of categorization is too harsh. A solution that keeps
+     and enhances the categorization is way more preferrable than one that nukes
+     it completely.
+
+   - As said before, the categorization is already present; this RFC proposes to
+     expose it to a higher level, in a structured, more discoverable format.
+
+   - Categorization is very traditional among software collections. Many of them
+     are doing this just fine for years on end, and Nixpkgs can imitate them
+     easily - and even surpass them, given the benefits of Nix language
+     machinery.
+
+   - Categorization is useful in many scenarios and use cases - indeed they
+     are ubiquitous in software world:
+     - specialized search engines (from Repology to MELPA)
+     - code forges, from Sourceforge to Gitlab
+     - as said above, software collections from pkgsrc to slackbuilds
+     - to organization and preservation (as Software Heritage)
+
+# Prior art
+[prior-art]: #prior-art
+
+As said above, categorization is very traditional among software collections. It
+is not hard to cite examples in this arena; the most interesting ones I have
+found are listed above (linked at [references section](#references))
+
+- FreeBSD Ports;
+- Debtags;
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Still unsolved is what data structure is better suited to represent a category.
+
+- For now we stick to a set `{ name, description }`.
+- Given the redundancy of the option above, another possibility is something
+  like `nameOfCategory = { description = ""; . . . }`
+
+# Future work
+[future]: #future-work
+
+- Curate the categories.
+- Update documentation.
+- Update Continuous Integration.
```

Comment:
Agree that it would be nice, unsure if it needs spelling out explicitly as a part of «Update CI». 

Or maybe some version of this could go into motivation?